cmd/minikube/cmd/start.go

```diff
@@ -1060,6 +1064,35 @@ func validateRegistryMirror() {
 	}
 }
 
+// This function validates that the --insecure-registry follows one of the following formats:
+// "<ip>:<port>" "<hostname>:<port>" "<network>/<netmask>"
+func ValidateInsecureRegistry() {
+	if len(insecureRegistry) > 0 {
+		validHostPortRegex := "^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]{0,61}[a-zA-Z0-9])(\\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]{0,61}[a-zA-Z0-9]))*$"
```

Comment:
I recommend making this a var package:

```
// validHostPortRegex is designed to pass these examples: 
var validHostPortRegex = regexp.MustCompile(`xxxx`)
```

Reply:
I think that while technically correct, this regexp is excessive, particularly given the abilities of re2, and the idea that not all DNS hosts match the RFC text. I suggest:

`^[\w\.-]$`

Reply:
For future use, I like to use this re2 tester: https://regoio.herokuapp.com/